--- a/libavfilter/cuda/vector_helpers.cuh
+++ b/libavfilter/cuda/vector_helpers.cuh
@@ -109,4 +109,27 @@
     );
 }
 
+template<typename T>
+inline __device__ T saturate_rintf(T a, float factor) {
+    return rintf(__saturatef(a) * factor);
+}
+
+template<>
+inline __device__ float2 saturate_rintf<float2>(float2 a, float factor) {
+    return make_float2(
+        saturate_rintf(a.x, factor),
+        saturate_rintf(a.y, factor)
+    );
+}
+
+template<>
+inline __device__ float4 saturate_rintf<float4>(float4 a, float factor) {
+    return make_float4(
+        saturate_rintf(a.x, factor),
+        saturate_rintf(a.y, factor),
+        saturate_rintf(a.z, factor),
+        saturate_rintf(a.w, factor)
+    );
+}
+
 #endif
--- a/compat/cuda/cuda_runtime.h
+++ b/compat/cuda/cuda_runtime.h
@@ -182,6 +182,7 @@
 static inline __device__ float fabs(float a) { return __builtin_fabsf(a); }
 static inline __device__ double fabs(double a) { return __builtin_fabs(a); }
 static inline __device__ float sqrtf(float a) { return __builtin_sqrtf(a); }
+static inline __device__ float rintf(float a) { return __builtin_rintf(a); }
 
 static inline __device__ float __saturatef(float a) { return __nvvm_saturate_f(a); }
 static inline __device__ float __sinf(float a) { return __nvvm_sin_approx_f(a); }
